sh_stm: Change stmmac driver so that it can operate when no phy is specified

This patch changes the stmmac driver so that it can operate when no PHY is
configured as is the case with the HDref platform.

Signed-off-by: Jon Frosdick <jon.frosdick@st.com>

Index: linux/drivers/net/stmmac/stmmac_main.c
===================================================================
--- linux.orig/drivers/net/stmmac/stmmac_main.c	2008-08-07 14:11:56.000000000 +0100
+++ linux/drivers/net/stmmac/stmmac_main.c	2008-08-07 14:15:41.000000000 +0100
@@ -248,6 +248,9 @@
 	int new_state = 0;
 	unsigned int fc = lp->flow_ctrl, pause_time = lp->pause;
 
+        if (phydev == NULL)
+           return;
+
 	DBG(probe, DEBUG, "stmmac_adjust_link: called.  address %d link %d\n",
 	    phydev->addr, phydev->link);
 
@@ -346,6 +349,11 @@
 	lp->speed = 0;
 	lp->oldduplex = -1;
 
+        if (lp->phy_addr == -1) {
+                /* We don't have a PHY, so do nothing */
+                return 0;
+        }
+
 	snprintf(phy_id, BUS_ID_SIZE, PHY_ID_FMT, lp->bus_id, lp->phy_addr);
 	DBG(probe, DEBUG, "stmmac_init_phy:  trying to attach to %s\n", phy_id);
 
@@ -1128,7 +1136,8 @@
 		lp->mac_type->ops->dump_dma_regs(ioaddr);
 	}
 
-	phy_start(lp->phydev);
+        if (lp->phydev)
+	   phy_start(lp->phydev);
 
 	netif_start_queue(dev);
 	return 0;
